### PR TITLE
Add SonarQube service endpoint

### PR DESCRIPTION
**Pull request by billbirchatcoles**
This PR adds a resource for SonarSource SA SonarQube service endpoints. 

## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

None 

Issue Number: #195

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

File: azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_sonaqube.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	r.Schema["url"] = &schema.Schema{
+		Type:         schema.TypeString,
+		Required:     true,
+		ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+		Description:  "Url for the SonarQube Server",
+	}
+
+	r.Schema["token"] = &schema.Schema{
+		Type:             schema.TypeString,
+		Optional:         true,
```
Comment (xuzhang3) [inline azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_sonaqube.go]: `token` should be `required`

File: website/docs/r/serviceendpoint_sonarqube.html.markdown [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+resource "azuredevops_serviceendpoint_sonarqube" "serviceendpoint" {
+
+  project_id        = azuredevops_project.project.id
+  service_endpoint_name = "sonarqube-mainframe"
+  description           = "My Service Connection"
+  url                   = "https://sonarqube.my.com"
+  token                 = "REDACTEDREDACTED"
+}
```
Comment (xuzhang3) [inline website/docs/r/serviceendpoint_sonarqube.html.markdown]: I would suggest change the example configuration to :
```suggestion
resource "azuredevops_serviceendpoint_sonarqube" "serviceendpoint" {

  project_id        = azuredevops_project.project.id
  service_endpoint_name = "Sample SonarQube"
  url                   = "https://sonarqube.my.com"
  token                 = "0000000000000000000000000000000000000000"
  description           = "Managed by Terraform"
}
```

**Review by xuzhang3 — COMMENTED** [review]
Few changes requested but otherwise looks good

File: website/docs/r/serviceendpoint_sonarqube.html.markdown (hunk 2 above)
Comment (birchb1024) [reply website/docs/r/serviceendpoint_sonarqube.html.markdown]: Done

File: azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_sonaqube.go (hunk 1 above)
Comment (birchb1024) [reply azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_sonaqube.go]: Done

**Comment by xuzhang3** [comment]
Hi  @billbirchatcoles   this branch has conflict with the mater, can you help handle the conflict?

**Comment by birchb1024** [comment]
Merged in latest master (3bd4c542). Conflict resolved. Retested.

**Comment by xuzhang3** [comment]
LGTM

**Pull request merged**